protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -584,6 +590,48 @@ definitions:
       - overwrite
       #- upsert_dedup # TODO chris: SCD Type 1 can be implemented later
       - append_dedup # SCD Type 1 & 2
+      - insert
+      - update
+      - upsert
+      - soft_delete
```

Comment:
Just curious: would we never need a `delete`? Wouldn't it kind of be up to the destination whether it does soft or hard delete operations?

Reply:
This is a good question. If we check Salesforce, it supports the five following operations:
* insert
* update
* upsert
* archive (soft delete)
* delete

So in theory, this is possible. I think the reasoning is that it is mostly scary to delete stuff altogether without the capability to revert so I think in most cases, people will like the have the ability to revert. Another datapoint that supports that is that [the HubSpot API](https://developers.hubspot.com/docs/reference/api/crm/objects/contacts#batch) doesn't support delete and only has the operations listed in this PR.